Data/Scripts/011_Battle/001_Battle/003_Battle_ExpAndMoveLearning.rb

```diff
@@ -162,7 +162,7 @@ def pbGainExpOne(idxParty, defeatedBattler, numPartic, expShare, expAll, showMes
     expGained = expFinal - pkmn.exp
     return if expGained <= 0
     # "Exp gained" message
-    if showMessages
+    if showMessages && !expAll
```

Comment:
I don't think this should be added. It looks like you're relying on the "Your Pokémon gained Exp. Points!" message (line 47) to cut down on the number of messages shown, but that message is only shown if the player has a non-participant in their party, which isn't necessarily the case. If they don't, but the player does have an Exp. All, no messages about Exp. gain will be shown at all, which is obviously not desirable.